src/users/users.controller.ts

```diff
@@ -13,11 +15,18 @@ export class UsersController {
         private usersService: UsersService,
     ) {}
 
+    // GET /users
+    // This action returns a user's profile
+    @Get()
+    async getProfile(@GetUser() user: UserDecoratorType): Promise<SanitizedUser> {
+        return await this.usersService.findOne({ id: user.sub });
+    }
+
     // GET /users/:username
     // This action returns a #${username} user
     @Get(':username')
-    async findOne(@Param('username') username: string, @Req() req: Request): Promise<SanitizedUser> {
-        if (req['user']?.username !== username) {
+    async findOne(@Param('username') username: string, @GetUser() user: UserDecoratorType): Promise<SanitizedUser> {
+        if (!user || (user.username !== username && !user.isAdmin)) {
             throw new UnauthorizedException();
         }
         return await this.usersService.findOne({ username });
```

Comment:
Suggestion: leave this route as `@AdminOnly()` to fetch a single user. Users could use the repository standard of `/users/my` to get their profile information. This is the current case for `/sshkeys/my`.

Reply:
Ok